tests/test_demo_flow.py

```diff
@@ -0,0 +1,111 @@
+        app.set_script('app/script.py', self.script_key)
+
+        # Create a tensor from the first 10 hashes by using the script (store it in redis)
+        self.dg.generate_data()
+        hashes_keys = self.redis_conn.zrange(self.references_key, 0, 9)
+        assert (len(hashes_keys) == 10)
+        output_key = 'out_tensor{tag}'
+        self.redis_conn.scriptexecute(self.script_key, 'hashes_to_tensor', keys=hashes_keys, outputs=[output_key])
+
+        # get the result and verify it's meta data
+        result = self.redis_conn.tensorget(output_key, meta_only=True)
+        assert (result['dtype'] == 'FLOAT')
+        assert (result['shape'] == [1, 256])
+
+    def test_entire_flow_multiple_clients(self):
+        app = app_runner.FraudDetectionApp(self.redis_conn)
+        app.set_script('app/script.py', 'helper_script{tag}')
+        app.set_model('app/models/creditcardfraud.pb', 'fraud_detection_model{tag}')
+        self.dg.generate_data()
+
+        def run_demo(output_key_name):
+            # Create a random transaction tensor in Time=100
+            transaction_tensor = np.random.randn(1, 30).astype(np.float32)
+            transaction_tensor[0] = 100
+
+            # Find the relevant reference data (transaction that occurred within the past 60 seconds)
+            ref_data_keys = self.redis_conn.zrangebyscore(self.references_key, 40, 100)
+
+            # Create a DAG (execution plan) for RedisAI. First, use the helper script to convert the reference data
+            # within the hashes into a tensor. Then run the model twice and obtain 2 outputs,
+            # and finally use the helper script to take their average to be the result (and persist it in key space)
+            dag = self.redis_conn.dag(persist=[output_key_name])
+            dag.tensorset('transaction', transaction_tensor)
+            dag.scriptexecute(self.script_key, 'hashes_to_tensor', keys=ref_data_keys, outputs=['reference'])
+            dag.modelexecute(self.model_key, inputs=['transaction', 'reference'], outputs=['out_1'])
+            dag.modelexecute(self.model_key, inputs=['transaction', 'reference'], outputs=['out_2'])
+            dag.scriptexecute(self.script_key, 'post_processing', inputs=['out_1', 'out_2'], outputs=[output_key_name])
+            dag.execute()
+
+        # run the demo scenario from multiple clients
+        n_clients = 10
+        clients = []
+        for i in range(n_clients):
+            client_output = 'result{tag}' + str(i)
+            p = Process(target=run_demo(client_output))
+            p.start()
+            clients.append(p)
+        # Wait for all clients to finish
+        [c.join() for c in clients]
+
+        # assert valid results
+        for i in range(n_clients):
+            client_output = 'result{tag}' + str(i)
+            result = self.redis_conn.tensorget(client_output)
+            assert (len(result[0]) == 2)
+            assert (0 <= result[0][0] <= 100 and 0 <= result[0][1] <= 100)
```

Comment:
please use redisai client and compare np tensors

Reply:
I am using redisai client... about using np for compare - will do.